kyo-combinators/shared/src/main/scala/kyo/AbortCombinators.scala

```diff
@@ -0,0 +1,637 @@
+package kyo
+
+import kyo.debug.Debug
+import kyo.kernel.ArrowEffect
+import scala.annotation.tailrec
+import scala.annotation.targetName
+import scala.util.NotGiven
+
+extension [A, S, E](effect: A < (Abort[E] & S))
+
+    /** Handles the Abort effect and returns its result as a `Result[E, A]`.
+      *
+      * @return
+      *   A computation that produces the result of this computation with the Abort[E] effect handled
+      */
+    def result(
+        using
+        ct: SafeClassTag[E],
+        fl: Flat[A],
+        fr: Frame
+    ): Result[E, A] < S =
+        Abort.run[E](effect)
+
+    /** Handles the Abort effect and returns its result as a `Result.Partial[E, A]`, not handling Panic exceptions.
+      *
+      * @return
+      *   A computation that produces a partial result of this computation with the Abort[E] effect handled
+      */
+    def partialResult(
+        using
+        ct: SafeClassTag[E],
+        fl: Flat[A],
+        fr: Frame
+    ): Result.Partial[E, A] < (Abort[Nothing] & S) =
+        Abort.runPartial(effect)
+
+    /** Handles the Abort effect and returns its result as a `Result.Partial[E, A]`, throwing Panic exceptions
+      *
+      * @return
+      *   A computation that produces a partial result of this computation with the Abort[E] effect handled
+      */
+    def partialResultOrThrow(
+        using
+        ct: SafeClassTag[E],
+        fl: Flat[A],
+        fr: Frame
+    ): Result.Partial[E, A] < S =
+        Abort.runPartialOrThrow(effect)
+
+    /** Handles the Abort effect, transforming caught errors into a new error as determined by mapping function
+      *
+      * @return
+      *   A computation that fails with Abort[E1], where E1 is an error type mapped from E
+      */
+    def mapAbort[E1, S1](
+        fn: E => E1 < S1
+    )(
+        using
+        ct: SafeClassTag[E],
+        ct1: SafeClassTag[E1],
+        fl: Flat[A],
+        fr: Frame
+    ): A < (Abort[E1] & S & S1) =
+        effect.catching(e => fn(e).map(Kyo.fail))
+
+    def forAbort[E1 <: E]: ForAbortOps[A, S, E, E1] = ForAbortOps(effect)
+
+    /** Translates the Abort effect to a Choice effect by handling failures as empty choices.
+      *
+      * @return
+      *   A computation that produces the result of this computation with the Abort[E] effect translated to an empty Choice
+      */
+    def abortToEmpty(
```

Comment:
From the name, I was expecting this to leave a pending `Abort[Empty]`. Maybe a more explicit name like `abortToChoiceDrop`?

Reply:
Yeah absent vs empty is confusing. I'm not sure about `...ChoiceDrop`, but I can't think of anything better at the moment.